tools/bdf-to-mono/src/lib.rs

```diff
@@ -34,32 +27,108 @@ pub fn bdf_to_bitmap(font: &BdfFont, encoding: Encoding) -> Result<Bitmap> {
         std::char::from_u32(*start).unwrap()..std::char::from_u32(*start + 16).unwrap()
     });
 
+    let mut glyph_bb = font
+        .glyphs
+        .get('A')
+        .ok_or_else(|| anyhow!("font doesn't contain 'A' glyph"))?
+        .bounding_box;
+    let mut fallback_glyph = None;
+    // used to optimize the glyph height in case all glyph offsets are e.g. -1
+    let mut max_y_offset = i32::MIN;
+    // We should take the DWIDTH value into account. If all glyph bounding boxes are smaller than
+    // DWIDTH the CHARACTER_SPACING in the MonoFont impl should be set to DWIDTH - (width of the BB)
+    // to maintain the correct kerning.
+    let mut char_spacing = None;
+
+    // find the maximum bounding box that can fit all glyphs taking offsets into account
+    // glyph_bb.offset.y will contain the baseline
+    // additionally build the `selected_glyphs` vector to save on the glyph lookup later in the code
+    let selected_glyphs = chars
+        .into_iter()
+        .enumerate()
+        .map(|(index, c)| {
```

Comment:
Here is an untested snippet how I would implement this:
```suggestion
    let glyph = chars
        .iter()
        .map(|c| {
            font.glyphs.get(c)
            .or_else(|| font.glyphs.get('?'))
            .ok_or_else(|| anyhow!("font doesn't contain the '{}' glyph and no fallback '?' glyph", c));
        })
        .collect::<Result<Vec<_>, _>>()?;
        
   /// TODO: calculate BB
```

IMO the code is a little easier to understand if you split the generation of the `selected_glyphs` vector and the bounding box calculation.

This would also replace missing `DEL` glyphs with the fallback glyph, which is a different behavior than before, but IMO this is also a valid replacement.

Reply:
Got it, fixed. Now, with the different handling of missing `DEL` char, the built-it fonts have changed (there is now the question mark in place of `DEL`). I'm not sure whether I should include the new .bin and .png files or not.

Reply:
> Got it, fixed.

Looks good, thanks for your patience with the comments.

> I'm not sure whether I should include the new .bin and .png files or not.

They should always reflect the current state of the font conversion tooling and therefore should be included in this PR.

Reply:
I've updated the built-in fonts